homeassistant/components/zha/core/device.py

```diff
@@ -157,17 +147,17 @@ def add_cluster_listener(self, cluster_listener):
         """Add cluster listener to device."""
         # only keep 1 power listener
         if cluster_listener.name is LISTENER_BATTERY and \
-                LISTENER_BATTERY in self._cluster_listeners:
+                LISTENER_BATTERY in self.cluster_listeners:
             return
         self._all_listeners.append(cluster_listener)
         if isinstance(cluster_listener, EventRelayListener):
             self._relay_listeners.append(cluster_listener)
         else:
-            self._cluster_listeners[cluster_listener.name] = cluster_listener
+            self.cluster_listeners[cluster_listener.name] = cluster_listener
 
     def get_cluster_listener(self, name):
```

Comment:
This also looks redundant. ☺

Reply:
how so?

Reply:
We can access the instance attribute that holds the dict directly.

Reply:
Ah you meant the get_cluster_listener function... the git difs with the review comments aren’t always clear. I thought you were talking about line 156 pictured in the dif! Makes sense now. Will remediate.